envelope_test.go

```diff
@@ -917,3 +917,51 @@ func TestEnvelopeHeaders(t *testing.T) {
 		}
 	}
 }
+
+func TestMissMatchedTransferEncoding(t *testing.T) {
+	headers := map[string]string{
+		"To":                        "cgarrett@stuff.com",
+		"Message-Id":                "<20170123170028.510AC13AE@slack-job-worker626.tinyspeck.com>",
+		"Mime-Version":              "1.0",
+		"From":                      `"Slack" <feedback@slack.com>`,
+		"Subject":                   "=?utf-8?Q?=5bSlack=5d_Stuffs_Company_updates_for_the_week_of_January_=31=35th=2c_=32=30=31=37?=",
+		"Content-Type":              `multipart/alternative; boundary="__slack_1231448102__"`,
+		"Date": "Mon, 23 Jan 2017 09:00:28 -0800 (PST)",
+	}
+
+	msg := openTestData("mail", "mime-missmatched-encoding.raw")
+	e, err := ReadEnvelope(msg)
+
+	if err != nil {
+		t.Fatal("Failed to parse MIME:", err)
+	}
+
+	if e.Text != "Stuff" {
```

Comment:
"Stuff" is not the expected text portion of this message, it should be the ~60 lines of text from the `Content-Type: text/plain; charset="utf-8"` MIME part.

What's happening here is enmime saw that envelope.Text was empty, and down-converted the HTML section to text.  That's not what we want.